matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/DeactivateLiveLocationShareWorker.kt

```diff
@@ -0,0 +1,95 @@
+import com.squareup.moshi.JsonClass
+import io.realm.RealmConfiguration
+import org.matrix.android.sdk.internal.SessionManager
+import org.matrix.android.sdk.internal.database.awaitTransaction
+import org.matrix.android.sdk.internal.database.model.livelocation.LiveLocationShareAggregatedSummaryEntity
+import org.matrix.android.sdk.internal.database.query.getOrCreate
+import org.matrix.android.sdk.internal.di.SessionDatabase
+import org.matrix.android.sdk.internal.session.SessionComponent
+import org.matrix.android.sdk.internal.worker.SessionSafeCoroutineWorker
+import org.matrix.android.sdk.internal.worker.SessionWorkerParams
+import timber.log.Timber
+import javax.inject.Inject
+
+/**
+ * Worker dedicated to update live location summary data so that it is considered as deactivated.
+ * For the context: it is needed since a live location share should be deactivated after a certain timeout.
+ */
+internal class DeactivateLiveLocationShareWorker(context: Context, params: WorkerParameters, sessionManager: SessionManager) :
+        SessionSafeCoroutineWorker<DeactivateLiveLocationShareWorker.Params>(
+                context,
+                params,
+                sessionManager,
+                Params::class.java
+        ) {
+
+    @JsonClass(generateAdapter = true)
+    internal data class Params(
+            override val sessionId: String,
+            override val lastFailureMessage: String? = null,
+            val eventId: String,
+            val roomId: String
+    ) : SessionWorkerParams
+
+    @SessionDatabase
+    @Inject lateinit var realmConfiguration: RealmConfiguration
+
+    override fun injectWith(injector: SessionComponent) {
+        injector.inject(this)
+    }
+
+    override suspend fun doSafeWork(params: Params): Result {
+        return runCatching {
+            deactivateLiveLocationShare(params)
+        }.fold(
+                onSuccess = {
+                    Result.success()
+                },
+                onFailure = {
+                    Timber.e("failed to deactivate live, eventId: ${params.eventId}, roomId: ${params.roomId}")
+                    Result.failure()
+                }
+        )
+    }
+
+    private suspend fun deactivateLiveLocationShare(params: Params) {
+        awaitTransaction(realmConfiguration) { realm ->
+            val aggregatedSummary = LiveLocationShareAggregatedSummaryEntity.getOrCreate(
+                    realm = realm,
+                    roomId = params.roomId,
+                    eventId = params.eventId
+            )
+            aggregatedSummary.isActive = false
+        }
+    }
+
+    override fun buildErrorParams(params: Params, message: String): Params {
+        return params.copy(lastFailureMessage = params.lastFailureMessage ?: message)
+    }
+
+    companion object {
+        private const val WORK_NAME_PREFIX = "DeactivateLiveLocationWork-"
+
+        fun getWorkName(eventId: String, roomId: String) = "${WORK_NAME_PREFIX}$eventId-$roomId"
```

Comment:
I do not know if the length of the string passed for `uniqueWorkName` can be limited, but maybe better to be safe here:
Maybe call `.md5()`, so replace by:
```kotlin
        fun getWorkName(eventId: String, roomId: String): String {
            val hash = "$eventId$roomId".md5()
            return "DeactivateLiveLocationWork-$hash"
        }
```

(you can keep the const WORK_NAME_PREFIX if you prefer)